src/analysis/sampling/CHRR/chrrSampler.m

```diff
@@ -1,146 +1,118 @@
-end
-
-if nargin < 6 || isempty(minFlux)
-    toGetWidths = 1;
-else
-    toGetWidths = 0;
-end
-
-if nargin < 8 || isempty(useFastFVA)
-    useFastFVA = false;
-end 
-% End of define defaults
-% Preprocess model
-if toPreprocess
-    
-    % parse the model to get the meat
-    P = chrrParseModel(model);
-    
-    %check for width 0 facets to make sure we are full dimensional
-    %also check for feasibility
-    fprintf('Checking for width 0 facets...\n');
-    
-    if toGetWidths
-        % check if we can use fastFVA
-        if useFastFVA && SOLVERS.ibm_cplex.installed
-            % check if we can do parallel for fastFVA
-            v=ver;
-            PCT='Parallel Computing Toolbox';
-            if  any(strcmp(PCT,{v.Name}))
-                p = parcluster('local');
-                setWorkerCount(p.NumWorkers);
-            end
-            [minFlux, maxFlux] = fastFVA(model,100);
-        elseif useFastFVA && ~SOLVERS.ibm_cplex.installed
-            fprintf('IBM CPLEX is not installed, so `fastFVA` cannot be run. Using `fluxVariability` instead\n');
-            [minFlux, maxFlux] = fluxVariability(model);
-        else
-            [minFlux, maxFlux] = fluxVariability(model);
-        end
-    end
-    
-    eps_cutoff = 1e-7;
-    
-    isEq = (maxFlux - minFlux) < eps_cutoff;
-    eq_constraints = sparse(sum(isEq),size(P.A_eq,2));
-    eq_constraints(:,isEq) = speye(sum(isEq));
-    
-    P.A_eq = [P.A_eq; eq_constraints];
-    P.b_eq = [P.b_eq; minFlux(isEq)];
-    
-    %check to make sure P.A and P.b are defined, and appropriately sized
-    if (isfield(P,'A')==0 || isfield(P,'b')==0) || (isempty(P.A) || isempty(P.b))
-        %either P.A or P.b do not exist
-        error('You need to define both P.A and P.b for a polytope {x | P.A*x <= P.b}.');
-    end
-    
-    [num_constraints,dim] = size(P.A);
-    
-    if exist('numSkip')~=1 || isempty(numSkip)
-        numSkip=8*dim^2;
-    end
-    
-    fprintf('Currently (P.A, P.b) are in %d dimensions\n', dim);
-    
-    if size(P.b,2)~= 1 || num_constraints ~= size(P.b,1)
-        error('Dimensions of P.b do not align with P.A.\nP.b should be a %d x 1 vector.',num_constraints);
-    end
-    
-    if (isfield(P,'A_eq')==0 || isempty(P.A_eq)) && ...
-            (isfield(P,'b_eq')==0 || isempty(P.b_eq))
-        P.A_eq = [];
-        P.b_eq = [];
-    end
-    
-    %preprocess the polytope of feasible solutions
-    %restict to null space
-    %round via maximum volume ellipsoid
-    [roundedPolytope] = preprocess(P, toRound);
-end
-
-fprintf('Generating samples...\n');
-
-%now we're ready to sample
-samples = genSamples(roundedPolytope, numSkip, numSamples);
-
-% samples = genSamplesGaussian(roundedPolytope,numSkip,numSamples,100*ones(size(roundedPolytope.N,1),1),eye(size(roundedPolytope.N,1)));
-
-end
\ No newline at end of file
+function [samples, roundedPolytope] = chrrSampler(model, numSkip, numSamples, toRound, roundedPolytope, useFastFVA,optPercentage)
```

Comment:
@Bounciness , any reason for removing 2 output arguments here?

Reply:
No, that was a mistake, sorry! It was a simple fix, but I'm getting a permission denied error when I try to push to your fork. I've put the file here in case it's helpful, but let me know if should try to push it again.
[chrrSampler.zip](https://github.com/opencobra/cobratoolbox/files/1564614/chrrSampler.zip)

Reply:
That's ok ... May you push to your own fork by branching off as before?
```
$ git fetch laurent
$ git checkout -b pr-1031-3 laurent/pr-1031
$ git add <yourFile>
$ git commit -m "<ourMsg>"
$ git push origin pr-1031-3
```

Reply:
Okay it's done: https://github.com/Bounciness/cobratoolbox/tree/pr-1031-3

Thank you for being patient with my lack of ability with git.

Reply:
Thanks @Bounciness - I merged your branch onto this PR. 👍